sycl/source/detail/queue_impl.hpp

```diff
@@ -225,15 +190,7 @@ class queue_impl {
   /// \param AsyncHandler is a SYCL asynchronous exception handler.
   queue_impl(ur_queue_handle_t UrQueue, const ContextImplPtr &Context,
              const async_handler &AsyncHandler)
-      : MContext(Context), MAsyncHandler(AsyncHandler),
-        MIsInorder(has_property<property::queue::in_order>()),
-        MDiscardEvents(
-            has_property<ext::oneapi::property::queue::discard_events>()),
-        MIsProfilingEnabled(has_property<property::queue::enable_profiling>()),
-        MQueueID{
-            MNextAvailableQueueID.fetch_add(1, std::memory_order_relaxed)} {
-    queue_impl_interop(UrQueue);
-  }
+      : queue_impl(UrQueue, Context, AsyncHandler, {}) {}
```

Comment:
Using delegating ctor here instead of non-ctor helper `queue_impl_interop` (that was inlined into the ctor below).